jobs/payment-jobs/tasks/ap_task.py

```diff
@@ -59,9 +63,58 @@ def create_ap_files(cls):
         6. Create AP file and upload to SFTP.
         7. After some time, a feedback file will arrive and Payment-reconciliation queue will set disbursement
            status to COMPLETED or REVERSED (filter out)
+        8. Find all EFT refunds with status REFUND_REQUESTED.
+        9. Create AP file and upload to SFTP.
+        10. After some time, a feedback file will arrive and Payment-reconciliation queue will move these
+            EFT refunds to REFUNDED. (filter out)
         """
         cls._create_routing_slip_refund_file()
         cls._create_non_gov_disbursement_file()
+        cls._create_eft_refund_file()
+
+    @classmethod
+    def _create_eft_refund_file(cls):
+        """Create AP file for EFT refunds and upload to CGI."""
+        cls.ap_type = EjvFileType.EFT_REFUND
+        eft_refunds_dao: List[tuple[InvoiceModel, EFTRefundModel]] = db.session.query(InvoiceModel, EFTRefundModel) \
+            .join(EFTCreditInvoiceLinkModel, EFTCreditInvoiceLinkModel.invoice_id == InvoiceModel.id) \
+            .join(EFTCreditModel, EFTCreditModel.id == EFTCreditInvoiceLinkModel.eft_credit_id) \
+            .join(EFTShortnameLinksModel, EFTShortnameLinksModel.eft_short_name_id == EFTCreditModel.short_name_id) \
+            .join(EFTRefundModel, EFTRefundModel.short_name_id == EFTShortnameLinksModel.eft_short_name_id) \
+            .filter(EFTRefundModel.status == InvoiceStatus.REFUND_REQUESTED.value) \
```

Comment:
Need to filter out eft_refund.disbursement_status_code = UPLOADED ?

Reply:
EFTRefundModels.status probably shouldn't be compared to the InvoiceStatus.

It also looks like when we create the eft_refund in pay api it uses EFTCreditInvoiceStatus. EFT refunds has its own flow, I would suggest it has its own set of statuses.

Something similar to:

PENDING_APPROVAL
REJECTED
APPROVED